packages/first-boot-wizard/files/usr/lib/lua/firstbootwizard.lua

```diff
@@ -219,22 +250,31 @@ function read_configs()
 end
 
 -- Apply configuration for a new network ( used in ubus daemon)
-function apply_user_configs(configs)
+function apply_user_configs(configs, hostname)
+    local uci_cursor = uci.cursor()
     -- Mesh network name
     local name = configs.ssid
+    -- Format hostname
+    hostname = hostname or uci_cursor:get("lime", "system", "hostname")
     -- Save changes in lime-defaults
-    local uci_cursor = uci.cursor()
     uci_cursor:set("lime-defaults", 'wifi', 'ap_ssid', name)
     uci_cursor:set("lime-defaults", 'wifi', 'apname_ssid', name..'/%H')
     uci_cursor:set("lime-defaults", 'wifi', 'adhoc_ssid', 'LiMe.%H')
     uci_cursor:set("lime-defaults", 'wifi', 'ieee80211s_mesh_id', 'LiMe')
     uci_cursor:commit("lime-defaults")
-    -- Apply new configuration and reboot
+    -- Apply new configuration and setup hostname
     clean_lime_config()
     utils.execute("/rom/etc/uci-defaults/91_lime-config")
-    utils.execute("rm /etc/first_run")
-    os.execute("(( /usr/bin/lime-config && /usr/bin/lime-apply && reboot 0<&- &>/dev/null &) &)")
-    return { configs = configs }
+    uci_cursor:set("lime", 'system', 'hostname', hostname)
+    uci_cursor:commit('lime')
+    -- Start sharing lime-defaults
+    -- Apply new configuration
+    os.execute("/usr/bin/lime-config")
+    os.execute("/usr/bin/lime-apply")
```

Comment:
Why is lime-apply needed if we reboot just after?

Reply:
done in 8dc3f5ee3d782a71d0149ba6cf2095b4d86a6f92